### DISCUSSION TITLE
Shard exceeding the size available in a worker node

### DISCUSSION THREAD

**Discussion by anandsowm** [citusdata/citus#5799] [Q&A] [answered]
If the worker node is limited to certain storage size[ e.g 2TB in Azure], and when there is more data coming in for a particular shard, will there be error thrown on reaching the limit, or is there any mechanism to spill/extend the shard to another node? In the Azure [docs,](https://docs.microsoft.com/en-us/azure/postgresql/hyperscale/howto-troubleshoot-read-only) there are only 2 options possible when worker node becomes read-only when disk is almost full - either rebalance the data or drop some data.

**Comment by marcocitus** [comment, marked as answer]
It would behave in the same way as when Postgres runs out of disk space (writing to WAL fails). We are starting to work on shard split machinery, but it's not available yet.
